admin/tools/provision/cloudmanager.py

```diff
@@ -299,7 +305,8 @@ def _build_instance_name(self, instance_id):
         :param instance_id: instance id
         :return:            instance name
         """
-        if isinstance(instance_id, unicode):
+        # TODO: not sure if the logic here is doing what it is supposed to do
+        if isinstance(instance_id, basestring):
```

Comment:
ok. If the intent is to remove non-ascii characters then an `encode().decode()` should be enough without the test of "is a string" (you know it's a string).